[Saravanan Swetha] iP

swethacool opened the pull request.

Commit:
list shown

Commit:
Level 2
add and list

Commit:
Level 2
add input from users and lists them

Commit:
Level 0 - Rename, Greet, Exit

Commit:
Level 1 - Echo

Commit:
Level 2 - Add, List

Commit:
Level 3 - Mark as done

Commit:
Todos, Events, Deadlines

Commit:
commit issues

Hudou0420 commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        Task[] tasks = new Task[100]; // Fixed-size array for storing tasks
```
suggestion
maybe can use static final variable MAXTASKCOUNT = 100 instead of using magic number in this line of code

kagiura replied:
i think it's outside of the scope of this course for now but i wonder if there's a way to make this not fixed-size and dynamic instead.... 🤔

Hudou0420 commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+            String userInput = scanner.nextLine();
+
+
+            if (userInput.equalsIgnoreCase("bye")) {
+                System.out.println("EchoBox: 👋 Bye! Hope to see you again soon! 🚀");
+                break;
+            }
+
+
+            if (userInput.equalsIgnoreCase("Hello")) {
+                System.out.println("EchoBox: Hey there! 😊 How’s your day?");
+            } else if (userInput.equalsIgnoreCase("sad")) {
+                System.out.println("EchoBox: Don't be sad! Here's a hug 🤗");
+            } else if (userInput.equalsIgnoreCase("list")) {
+                System.out.println("Here are the tasks in your list: ");
+                for (int i = 0; i < taskCount; i++) { // Display task list
+                    System.out.println((i + 1) + ".[" + tasks[i].getStatusIcon() + "] " + tasks[i].description);
+                }
+            } else if (userInput.startsWith("unmark")) {
+                int taskNum = Integer.parseInt(userInput.split(" ")[1]) - 1;
+                if (taskNum >= 0 && taskNum < taskCount) {
+                    tasks[taskNum].unmarkAsDone();
+                    System.out.println("EchoBox: OK, I've marked this task as not done yet: ");
+                    System.out.println("[ ]" + tasks[taskNum].description);
+                } else {
+                    System.out.println("EchoBox: ⚠️ Invalid task number!");
+                }
+            } else if (userInput.startsWith("mark")) {
```
Good use of string methods (have learned from you)!

Hudou0420 commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        Task[] tasks = new Task[100]; // Fixed-size array for storing tasks
+        int taskCount = 0; // Counter for stored tasks
```
suggestion
maybe rename this to "taskCounter" for clearer meaning

Hudou0420 commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+                    System.out.println("EchoBox: Nice! I've marked this task as done: ");
+                    System.out.println("[X]" + tasks[taskNum].description);
+                } else {
+                    System.out.println("EchoBox: ⚠️ Invalid task number!");
```
Good use of symbols! Does it work well in the CLI?

kagiura commented on desktop.ini:
i don't think this file should be here? pls correct me if i'm wrong 🙏 could try putting it in gitignore

kagiura commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+
+
+public class EchoBox {
+    // Task class to represent a task
```
slightly nitpicky but i think these are a bit redundant, maybe better to remove since the class name is straightforward anyways

kagiura commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+        System.out.println("😊 How can I help you today?");
+
+
+        while (true) {
```
i feel like this could be extracted to a function? so the main function isn't too long / can be broken down a bit 🤔

kagiura commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+
+
+        System.out.println("Hello! I'm EchoBox"); // Your chatbot's new name
+        System.out.println("😊 How can I help you today?");
```
i love the use of emojis it makes it feel much more livelier haha

kagiura commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+            }
+
+
+            if (userInput.equalsIgnoreCase("Hello")) {
```
think its better to put "hello" in lowercase to keep it consistent 👍 great idea for using ignoreCase btw!

kagiura commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+            if (userInput.equalsIgnoreCase("Hello")) {
+                System.out.println("EchoBox: Hey there! 😊 How’s your day?");
+            } else if (userInput.equalsIgnoreCase("sad")) {
+                System.out.println("EchoBox: Don't be sad! Here's a hug 🤗");
```
🥺

Hudou0420 reviewed (commented):
Looks good to me

kagiura commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+            String userInput = scanner.nextLine();
+
+
+            if (userInput.equalsIgnoreCase("bye")) {
```
instead of a very long if else chain i think a switch statement would really help! and preferrably if all the sub-functions could be extracted out too perhaps? eg:
```
command = userInput.toLowercase();
switch (command) {
  case "bye":
    exitProgram();
  case "hello":
    greetUser();
    break;
// ....
}```

This way when you find a problem with a specific command you dont have to scroll through the entire file as well

kagiura reviewed (commented):
great first PR :D the code could be simplified down hopefully so its easier to deal w/ and expand in the future!

YanXinyaoo commented on src/main/java/EchoBox.java:
Maybe it is better not to leave so much space between different methods.

YanXinyaoo commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+                int taskNum = Integer.parseInt(userInput.split(" ")[1]) - 1;
+                if (taskNum >= 0 && taskNum < taskCount) {
+                    tasks[taskNum].unmarkAsDone();
+                    System.out.println("EchoBox: OK, I've marked this task as not done yet: ");
```
Maybe add some comments to explain what are you trying to achieve here so it is more clear.

YanXinyaoo commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+            }
+            else {
+                // Add new task if not "list", "mark", or "unmark"
+                if (taskCount < 100) {
```
Maybe it is better to use a variable like maxTaskNumber to store 100 so next time if you want to change the value you only need to change once and do not need to find all the "100" in your code.

YanXinyaoo commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+            } else if (userInput.startsWith("unmark")) {
+                int taskNum = Integer.parseInt(userInput.split(" ")[1]) - 1;
+                if (taskNum >= 0 && taskNum < taskCount) {
+                    tasks[taskNum].unmarkAsDone();
```
Good check for the boundary of the list!

kinneo commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+                    System.out.println((i + 1) + ".[" + tasks[i].getStatusIcon() + "] " + tasks[i].description);
+                }
+            } else if (userInput.startsWith("unmark")) {
+                int taskNum = Integer.parseInt(userInput.split(" ")[1]) - 1;
```
this line assumes valid input, but if the user enters an invalid command, it could throw a NumberFormatException. Consider wrapping it in a try-catch block.

kinneo commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+                for (int i = 0; i < taskCount; i++) { // Display task list
+                    System.out.println((i + 1) + ".[" + tasks[i].getStatusIcon() + "] " + tasks[i].description);
+                }
+            } else if (userInput.startsWith("unmark")) {
```
Magic string. Instead of hardcoding "list", "mark", "unmark", "bye", etc., in multiple places, define them as static final String constants for better maintainability and avoiding typos.

kinneo commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+
+public class EchoBox {
+    // Task class to represent a task
+    static class Task {
```
The Task class could have getter and setter methods instead of directly exposing description and isDone. This ensures better encapsulation and future flexibility.

kinneo commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,93 @@
+
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
```
The Scanner object is never closed. Since it takes input from System.in, it should be properly closed at the end of execution using scanner.close(); to prevent resource leaks.

kinneo reviewed (commented):
lgtm keep up the good work

Commit:
Todos, Events, Deadlines

Commit:
Add archive.zip and update EchoBox.java

Commit:
Merge branch 'master' of https://github.com/swethacool/ip

Commit:
A-TextUiTesting

Matching expected output to actual output to verify

Commit:
todos, events, deadlines

Commit:
Merge branch 'master' of https://github.com/swethacool/ip

Commit:
A-CodeQuality

-convert to switch statements as nested if-else statements are harder to read
-break down my main method into smaller sub methods to improve readiblity
-instead of hardcoding commands/numbers, I replaced it with symbolic constants

Commit:
Created EchoBoxException to handle errors

Commit:
-Handle errors

-exceptions included
-moved the Todo, Event, Deadline, Task methods into separate packages

Commit:
Merge branch-Level-5 into master

Commit:
Delete-Add support for deleting tasks from the list
Implemented A-Collections
Removed unnecessary spacings and added comments

Commit:
Level-7: Save

Commit:
Merge branch 'branch-Level-6' into master

Commit:
Merge Level-7 into master

Resolved merge conflicts

Commit:
A-Jar
Executable file

Paulifyer commented on src/main/java/EchoBox.java:
```diff
@@ -0,0 +1,182 @@
+                System.out.println("EchoBox: " + e.getMessage());  // Handle and print the exception message
+            }
+        }  else if (command.startsWith("unmark ")) {
+            try {
+                handleUnmark(userInput);  // This may throw EchoBoxException
+            } catch (EchoBoxException e) {
+                System.out.println("EchoBox: " + e.getMessage());  // Handle and print the exception message
+            }
+        } else if (command.startsWith("todo ")) {
+            try {
+                handleAddTodo(userInput);
+            } catch (EchoBoxException e) {
+                System.out.println("EchoBox: " + e.getMessage());
+            }
+        } else if (command.startsWith("deadline ")) {
+            try {
+                handleAddDeadline(userInput);
+            } catch (EchoBoxException e) {
+                System.out.println("EchoBox: " + e.getMessage());
+            }
+        } else if (command.startsWith("event ")) {
+            try {
+                handleAddEvent(userInput);
+            } catch (EchoBoxException e) {
+                System.out.println("EchoBox: " + e.getMessage());
+            }
+        } else if (command.startsWith("delete ")) {
+            try{
+                handleDelete(userInput);
+            } catch (EchoBoxException e) {
+                System.out.println("EchoBox: " + e.getMessage());
+            }
+        } else if (userInput.equalsIgnoreCase("Hello")) {
+            System.out.println("EchoBox: Hey there! How is your day?");
+        } else if (userInput.equalsIgnoreCase("sad")) {
+            System.out.println("EchoBox: Don't be sad! Here's a hug :)");
+        } else if(!userInput.equalsIgnoreCase(EXIT_COMMAND)){
+            System.out.println("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
```
Good use of intermediate methods to handle commands. Additionally, for better readability you could use a switch statement instead

Commit:
MoreOOP

Refactored the original code into Storage, Parser, TaskList, Ui classes with each holding various methods respectively for easier accessiblity and readiblity

Commit:
Level-9

Find: Giving users a way to find a task by searching for a keyword in the task description.

Commit:
JavaDoc

Add JavaDoc comments to the code

Commit:
A-MoreOOP

Refactored the code further by implementing Commands class

Commit:
A-MoreOOP

Refactored the code to implement the Command class
Added JavaDoc comments

Commit:
A-UserGuide

A user guide for EchoBox

Commit:
removed unknown submodule that caused build error

Commit:
A-UserGuide

final version after editing some changes
